src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -141,7 +145,46 @@ def docker_login_to_registry(login_server):
             "Access to registry was denied. Response code: {}".format(response.status_code))
 
     refresh_token = loads(response.content.decode("utf-8"))["refresh_token"]
+    if only_refresh_token:
+        return refresh_token, ''
 
+    authhost = urlunparse((authurl[0], authurl[1], '/oauth2/token', '', '', ''))
+    if repository is None:
+        scope = 'registry:catalog:*'
+    else:
+        scope = 'repository:' + repository + ':pull'
```

Comment:
Thanks, otherwise, this PR looks good to me. Please consider rebasing from the main repo before/after you merge, there might be some conflict in is_tty check part. /cc @SajayAntony